leases/schema/mutations.py

```diff
@@ -209,15 +210,23 @@ def mutate_and_get_payload(cls, root, info, **input):
                 _("Application must be connected to an existing customer first")
             )
 
-        place = get_node_from_global_id(
-            info,
-            input.pop("place_id"),
-            only_type=WinterStoragePlaceNode,
-            nullable=False,
-        )
+        if place_id := input.pop("place_id", None):  # noqa
+            place = get_node_from_global_id(
+                info, place_id, only_type=WinterStoragePlaceNode, nullable=False,
+            )
+            input["place"] = place
+            area = place.winter_storage_section.area
+        elif area_id := input.pop("area_id", None):  # noqa
+            area = get_node_from_global_id(
+                info, area_id, only_type=WinterStorageAreaNode, nullable=False,
+            )
+            input["area"] = area
+        else:
```

Comment:
Should we also check that "area" and "place" are not both defined at the same time?

Reply:
Right! Slipped out of my mind.